cdap-tms/src/main/java/co/cask/cdap/messaging/client/ClientRollbackDetail.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.messaging.client;
+
+import co.cask.cdap.messaging.RollbackDetail;
+import co.cask.cdap.messaging.Schemas;
+import com.google.common.base.Throwables;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.DecoderFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Client side implementation of {@link RollbackDetail}. It retains the original encoded bytes as it is most likely
+ * being used instead of the decoded form. The decoding happens on demand when the actual information is needed.
+ * However, the need for decoding should be rare and is not needed in normal operation.
+ */
+final class ClientRollbackDetail implements RollbackDetail {
+
+  private final byte[] encoded;
+  private GenericRecord decoded;
+
+  ClientRollbackDetail(byte[] encoded) {
+    this.encoded = encoded;
+  }
+
+  /**
+   * Returns the information contained in this class in the original encoded form as responded from the server.
+   */
+  byte[] getEncoded() {
+    return encoded;
+  }
+
+  @Override
+  public long getTransactionWritePointer() {
+    return (Long) getDecoded().get("transactionWritePointer");
```

Comment:
not required in this PR but should we have constants for these fields `transactionWritePointer`, `startTimestamp`, etc.. since these are used at multiple places and easy to make spelling mistakes.